app/src/sharedTest/java/org/oppia/android/app/databinding/ImageViewBindingAdaptersTest.kt

```diff
@@ -111,11 +138,78 @@ class ImageViewBindingAdaptersTest {
   fun setUp() {
     setUpTestApplicationComponent()
     Intents.init()
+    testCoroutineDispatchers.registerIdlingResource()
+    profileTestHelper.initializeProfiles()
   }
 
   @After
   fun tearDown() {
     Intents.release()
+    testCoroutineDispatchers.unregisterIdlingResource()
+  }
+
+  @Test
+  fun testImageViewBindingAdapters_imageView_has_correctDrawableByResourceId() {
+    activityRule.scenario.runWithActivity {
+      testCoroutineDispatchers.runCurrent()
+      val imageView = getImageView(it)
+      testCoroutineDispatchers.runCurrent()
+      setImageDrawableCompat(
+        imageView,
+        R.drawable.ic_portrait_onboarding_0
+      )
+      testCoroutineDispatchers.runCurrent()
+      onView(allOf(withId(R.id.image_view_for_data_binding)))
+        .check(matches(withDrawable(R.drawable.ic_portrait_onboarding_0)))
+      onView(isRoot()).perform(orientationLandscape())
+      onView(allOf(withId(R.id.image_view_for_data_binding)))
+        .check(matches(withDrawable(R.drawable.ic_portrait_onboarding_0)))
+    }
+  }
+
+  @Test
+  fun testImageViewBindingAdapters_imageView_has_correctDrawableByDrawable() {
+    activityRule.scenario.runWithActivity {
+      testCoroutineDispatchers.runCurrent()
+      val imageView = getImageView(it)
+      testCoroutineDispatchers.runCurrent()
+      val drawable = it.getDrawable(R.drawable.ic_portrait_onboarding_0)
+      testCoroutineDispatchers.runCurrent()
+      setImageDrawableCompat(imageView, drawable)
+      testCoroutineDispatchers.runCurrent()
+      onView(allOf(withId(R.id.image_view_for_data_binding)))
+        .check(matches(withDrawable(R.drawable.ic_portrait_onboarding_0)))
+      testCoroutineDispatchers.runCurrent()
+      onView(isRoot()).perform(orientationLandscape())
+      testCoroutineDispatchers.runCurrent()
+      onView(allOf(withId(R.id.image_view_for_data_binding)))
+        .check(matches(withDrawable(R.drawable.ic_portrait_onboarding_0)))
+      testCoroutineDispatchers.runCurrent()
+    }
+  }
+
+  @Test
+  fun testImageViewBindingAdapters_imageView_setProfileImage() {
+    activityRule.scenario.runWithActivity {
+      val imageView = getImageView(it)
+      val uri =
+        "black-man-male-short-hair-beard-hand-drawn-illustration_5362391.html"
+      val profileAvatar = ProfileAvatar.newBuilder()
+        .setAvatarImageUri("https://pngtree.com/freepng/$uri")
+        .build()
+      val url = profileAvatar.avatarImageUri.toString()
+      setProfileImage(imageView, profileAvatar)
+      testCoroutineDispatchers.runCurrent()
+      lateinit var imageBitmapSource: ImageTarget<Bitmap>
```

Comment:
This variable remains uninitialized and throws error

Reply:
You're never assigning this to a value--that's why it's uninitialized.

What, specifically, are you trying to accomplish here? This looks quite different from other tests in the codebase, so I'm wondering if you're trying to do something specific that could be solved with patterns used in other tests. Perhaps look for a similar test elsewhere and model this after one of those.

Reply:
I have changed the implementation.
Earlier I thought that loadBitmap(param1,param2)
param1 is the url and param2 is the target variable so I made the variable lateinit and the next line was supposed to initialize it but it never happened.
So now I'm creating a dynamic image view in which i'm loading the image with loadBitmap and then I'll access the BitmapDrawable and pass it to the test function.